authors/apps/articles/views.py

```diff
@@ -99,3 +100,56 @@ def retrieve(self, request, slug):
         serializer = self.detailSerializer()
         serializer.instance = article_instance
         return Response(serializer.data, status=status.HTTP_200_OK)
+
+class FavoriteArticles(generics.ListCreateAPIView):
+
+    queryset = FavoriteArticle.objects.all()
+
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+
+    serializer_class = FavoriteArticlesSerializer
+
+    lookup_url_kwarg = 'slug'
+
+    def post(self, request, *args, **kwargs):
+
+        slug = self.kwargs.get(self.lookup_url_kwarg)
+
+        article_instance = Article.objects.get(slug=slug)
+
+        article_id = article_instance.id
+
+        article_favorite = request.data.get('favorite_status')
+
+        token = request.META.get('HTTP_AUTHORIZATION', " ").split(' ')[1]
+
+        decoded_token = jwt.decode(token, SECRET_KEY, algorithm='HS256')
+
+        username = decoded_token['username']
+
+        user_instance = User.objects.get(username=username)
+
+        user_id = user_instance.id
+
+
+        favoriting_body = {
+
+            "article": article_id,
+
+            "favorite_status": article_favorite,
+
+            "author": user_id
+        }
+   
+        serializer = self.serializer_class(data=favoriting_body)
+
+        serializer.is_valid(raise_exception=True)
+
+        serializer.save()
+
+        data = serializer.data
+        
+        data["message"] = "You Have Successfully Added This Article To Your Favorites"
```

Comment:
It would also be better to do this sort of data shaping in the render's since you already have one.